coverage_comment/coverage.py

```diff
@@ -82,10 +85,26 @@ class DiffCoverage:
     files: dict[pathlib.Path, FileDiffCoverage]
 
 
-def compute_coverage(num_covered: int, num_total: int) -> decimal.Decimal:
-    if num_total == 0:
+def compute_coverage(
+    num_covered: int,
+    num_total: int,
+    use_branch_coverage: bool,
+    num_branches_covered: int | None,
+    num_branches_total: int | None,
+) -> decimal.Decimal:
+    """Compute the coverage percentage, with or without branch coverage."""
+    if use_branch_coverage:
+        num_branches_covered = cast(int, num_branches_covered)
+        num_branches_total = cast(int, num_branches_total)
+        numerator = decimal.Decimal(num_covered + num_branches_covered)
+        denominator = decimal.Decimal(num_total + num_branches_total)
```

Comment:
This math exactly checks out, with the `raw_data.totals` numbers and `raw_data.totals.percent_covered` produced by pytest.